### ISSUE TITLE
exp10() with Float16

### ISSUE THREAD

**Issue by andferrari** [JuliaLang/julia#27848]
It seems that `exp10()` is not defined for Float16
```julia
julia> exp10(Float16(1.1))
ERROR: StackOverflowError:
Stacktrace:
 [1] exp10(::Float16) at ./math.jl:295 (repeats 80000 times)
```

```julia
julia> versioninfo()
Julia Version 0.6.3
Commit d55cadc350 (2018-05-28 20:20 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, broadwell)
```

**Comment by JeffBezanson** [comment]
Seems to be fixed on master.

**Comment by StefanKarpinski** [comment]
Could possibly use a backport? Unless it's an LLVM miscompilation, which would be a bit hard to fix.

**Comment by johnfgibson** [comment]
The fix on master was backported to 0.6.4 in #27621, so this should be closed. 

```julia
julia> exp10(Float16(0.0))
Float16(1.0)

julia> versioninfo()
Julia Version 0.6.4-pre.13
Commit b72bc45efb (2018-07-04 17:37 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i7-3960X CPU @ 3.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge MAX_THREADS=16)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, sandybridge)
```
